src/vs/workbench/contrib/search/browser/searchModel.ts

```diff
@@ -2004,21 +2130,33 @@ export class SearchModel extends Disposable {
 	}
 
 	async notebookSearch(query: ITextQuery, token: CancellationToken, onProgress?: (result: ISearchProgressItem) => void): Promise<{ completeData: ISearchComplete; scannedFiles: ResourceSet }> {
+		const searchStart = Date.now();
 		const localResults = await this.getLocalNotebookResults(query, token);
+		const searchLocalEnd = Date.now();
+
+		const experimentalNotebooksEnabled = this.configurationService.getValue<ISearchConfigurationProperties>('search').experimental?.closedNotebookRichContentResults ?? false;
+
+		let closedResults = undefined;
+		if (experimentalNotebooksEnabled) {
+			const locallyScannedNotebookFiles = new ResourceSet([...localResults.results.keys()], uri => this.uriIdentityService.extUri.getComparisonKey(uri));
+			closedResults = await this.getClosedNotebookResults(query, locallyScannedNotebookFiles, token);
```

Comment:
Can closed notebooks be searched in parallel with open notebooks?

Reply:
I think that it currently needs a list of notebooks that are already open, but that _shouldn't_ be too bad to provide ahead of time.

Reply:
The same applies to this as https://github.com/microsoft/vscode/pull/174287#discussion_r1245628564